privcount/tally_server.py

```diff
@@ -205,6 +207,58 @@ def refresh_config(self):
                 ts_conf['noise'] = {}
                 ts_conf['noise']['counters'] = conf['counters']
 
+            # if we are counting a traffic model
+            if 'traffic_model' in ts_conf:
+                # we need the model, which specifies which counters we need to count
+                # make sure the model file exists
+                ts_conf['traffic_model'] = normalise_path(ts_conf['traffic_model'])
+                assert os.path.exists(ts_conf['traffic_model'])
+
+                # import and validate the model
+                with open(ts_conf['traffic_model'], 'r') as fin:
+                    traffic_model_conf = json.load(fin)
+                    assert check_traffic_model_config(traffic_model_conf)
+
+                # store the configs so we can transfer them later
+                ts_conf['traffic_model'] = traffic_model_conf
+
+                # get an object and register the dynamic counters
+                tmodel = TrafficModel(traffic_model_conf)
+
+                # we also need noise parameters for all of the traffic model counters
+                # make sure the noise file exists
+                assert 'traffic_noise' in ts_conf
+                ts_conf['traffic_noise'] = normalise_path(ts_conf['traffic_noise'])
+                assert os.path.exists(ts_conf['traffic_noise'])
+
+                # import and validate the noise
+                with open(ts_conf['traffic_noise'], 'r') as fin:
+                    traffic_noise_conf = yaml.load(fin)
+                assert tmodel.check_noise_config(traffic_noise_conf)
+
+                # store the configs so we can transfer them later
+                ts_conf['traffic_noise'] = traffic_noise_conf
+
+                # supplying a traffic model implies that the tally server
+                # wants to enable all counters associated with that model
+                # register the dynamic counter labels that will be needed for this model
+                tmodel.register_counters()
+
+                # get the bins and noise that we should use for this model
+                tmodel_bins = tmodel.get_bins_init_config()
+                tmodel_noise = tmodel.get_noise_init_config(traffic_noise_conf)
+
+                # sanity check
+                if set(tmodel_bins.keys()) != set(tmodel_noise.keys()):
```

Comment:
Hmm, `check_combined_counters()` seems to require sigmas, but at this point we have estimated values and sensitivities.

I think it's OK, because the next thing this function does is call `get_noise_allocation()` to compute the sigmas, and then it asserts that `check_counters_config()` succeeds. So we have another check already built in.